### PR TITLE
fix(session): DefaultConnectSessionFactory will failed if not in spring context

### PR DESCRIPTION
**Pull request by ungreat**
#### Which issue(s) this PR fixes:

Fixes #

#### Additional documentation e.g., usage docs, etc.:

```docs

```

### CODE REVIEW

File: server/odc-service/src/main/java/com/oceanbase/odc/service/monitor/datasource/GetConnectionFailedEventListener.java [hunk 1]
```diff
@@ -15,27 +15,22 @@
...
-        this.meterManager = meterManager;
-    }
+    final static EventStore eventStore = new EventStore(10000);
 
     @Override
     public void onEvent(GetConnectionFailedEvent event) {
-        meterManager.incrementCounter(MeterKey.ofMeter(DATASOURCE_GET_CONNECTION_FAILED_COUNT));
+        boolean offer = eventStore.offer(event);
+        if (!offer) {
+            log.error("Can't offer GetConnectionFailedEvent");
```
Comment (yhilmare) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/monitor/datasource/GetConnectionFailedEventListener.java:32]: log.warn will be better

File: server/odc-service/src/main/java/com/oceanbase/odc/service/monitor/datasource/ConnectFailedListenerDelegate.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+@Component
+@ConditionalOnProperty(value = "odc.system.monitor.actuator.enabled", havingValue = "true")
+class ConnectFailedListenerDelegate implements InitializingBean {
+
+    private final EventStore eventStore = GetConnectionFailedEventListener.eventStore;
+    private final ScheduledExecutorService scheduledExecutorService;
+    @Autowired
+    private MeterManager meterManager;
+
+    public ConnectFailedListenerDelegate() {
```
Comment (yhilmare) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/monitor/datasource/ConnectFailedListenerDelegate.java:46]: how about using `@Scheduled` and we can make spring manage the thread by this way

File: server/odc-service/src/main/java/com/oceanbase/odc/service/monitor/datasource/EventStore.java [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+    public EventStore(int size) {
+        eventQueue = new LinkedBlockingQueue<>(size);
+    }
+
+    public boolean offer(GetConnectionFailedEvent event) {
+        return eventQueue.offer(event);
+    }
+
+    public List<GetConnectionFailedEvent> eventDrainTo(int size) {
```
Comment (yhilmare) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/monitor/datasource/EventStore.java:37]: if there should be concurrent condition?

File: server/odc-service/src/main/java/com/oceanbase/odc/service/monitor/datasource/ConnectFailedListenerDelegate.java (hunk 2 above)
Comment (ungreat) [reply server/odc-service/src/main/java/com/oceanbase/odc/service/monitor/datasource/ConnectFailedListenerDelegate.java]: That's good idea~

File: server/odc-service/src/main/java/com/oceanbase/odc/service/monitor/datasource/EventStore.java (hunk 3 above)
Comment (ungreat) [reply server/odc-service/src/main/java/com/oceanbase/odc/service/monitor/datasource/EventStore.java:37]: BlockingQueue ensures concurrency security

File: server/odc-service/src/main/java/com/oceanbase/odc/service/monitor/datasource/ConnectFailedListenerDelegate.java [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+import com.oceanbase.odc.core.datasource.event.GetConnectionFailedEvent;
+import com.oceanbase.odc.service.monitor.MeterKey;
+import com.oceanbase.odc.service.monitor.MeterManager;
+
+@Component
+@ConditionalOnProperty(value = "odc.system.monitor.actuator.enabled", havingValue = "true")
+class ConnectFailedListenerDelegate {
+
+    private static final Logger log = LoggerFactory.getLogger(ConnectFailedListenerDelegate.class);
```
Comment (yhilmare) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/monitor/datasource/ConnectFailedListenerDelegate.java:38]: why shouldn't we use `@Sl4j`?

File: server/odc-service/src/main/java/com/oceanbase/odc/service/monitor/datasource/ConnectFailedListenerDelegate.java [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+    private final EventStore eventStore = GetConnectionFailedEventListener.eventStore;
+
+    @Autowired
+    private MeterManager meterManager;
+
+    @Scheduled(fixedDelay = 1000)
```
Comment (yhilmare) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/monitor/datasource/ConnectFailedListenerDelegate.java:44]: please confirm that the Frequency

File: server/odc-service/src/main/java/com/oceanbase/odc/service/monitor/datasource/ConnectFailedListenerDelegate.java [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.stereotype.Component;
+
+import com.oceanbase.odc.core.datasource.event.GetConnectionFailedEvent;
+import com.oceanbase.odc.service.monitor.MeterKey;
+import com.oceanbase.odc.service.monitor.MeterManager;
+
+@Component
+@ConditionalOnProperty(value = "odc.system.monitor.actuator.enabled", havingValue = "true")
+class ConnectFailedListenerDelegate {
```
Comment (yhilmare) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/monitor/datasource/ConnectFailedListenerDelegate.java:36]: please confirm that if it is work if there is not `public` access?

File: server/odc-service/src/main/java/com/oceanbase/odc/service/monitor/datasource/ConnectFailedListenerDelegate.java (hunk 4 above)
Comment (ungreat) [reply server/odc-service/src/main/java/com/oceanbase/odc/service/monitor/datasource/ConnectFailedListenerDelegate.java]: use @Sl4j instead

**Review by yhilmare — APPROVED** [review]
lgtm

**Pull request merged**